infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protostream/ProtoDataMapper.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.impl.protostream;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.hibernate.ogm.datastore.infinispanremote.impl.AssociationCacheOperation;
+import org.hibernate.ogm.datastore.infinispanremote.impl.CacheOperation;
+import org.hibernate.ogm.datastore.infinispanremote.impl.ProtoStreamMappingAdapter;
+import org.hibernate.ogm.datastore.infinispanremote.impl.ProtostreamAssociationMappingAdapter;
+import org.hibernate.ogm.datastore.infinispanremote.impl.VersionedAssociation;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.infinispan.protostream.DescriptorParserException;
+import org.infinispan.protostream.SerializationContext;
+
+public final class ProtoDataMapper implements ProtoStreamMappingAdapter, ProtostreamAssociationMappingAdapter {
+
+	private final SerializationContext serContext;
+	private final MainOgmCoDec delegate;
+	private final OgmProtoStreamMarshaller marshaller;
+
+	public ProtoDataMapper(MainOgmCoDec delegate, SerializationContext serContext, OgmProtoStreamMarshaller marshaller) throws DescriptorParserException, IOException {
+		this.marshaller = marshaller;
+		this.delegate = Objects.requireNonNull( delegate );
```

Comment:
Here we have access to the actual marshaller and the serialization context. Couldn't this be used to initialize the marshaller for once with the right context instead of setting this via the thread local again and again?